fuse/eval/metrics/metrics_common.py

```diff
@@ -109,6 +109,19 @@ def collect(self, batch: Dict) -> None:
             batch = NDict(batch)
 
         samples = uncollate(batch)
+
+        # If in distributed mode (multi gpu training) we shall gather the result from all the machine to evaluate with respect to the entire batch.
+        if dist.is_initialized():
+            world_size = dist.get_world_size()  # num of gpus
+            samples_gather = [None for rank in range(world_size)]
+            # samples_gather[i] will have the 'samples' value of the i's GPU
+            dist.all_gather_object(samples_gather, samples)
```

Comment:
Interesting, Any idea how it knows from where to collect from? 
Shouldn't it be done only for the main GPU?

Reply:
I think it should be done in each GPU so the processes will transfer the data. And each process knows it's local rank (index in the array).

Not so sure so lets take it offline :)